### ISSUE TITLE
AWSMobileClient showSignIn() failure on iOS 14.5

### ISSUE THREAD

**Issue by doc-rj-celltrak** [aws-amplify/aws-sdk-ios#3586]
**Describe the bug**
Starting in iOS 14.5, there is a crash when calling showSignIn() and starting an ASWebAuthenticationSession. The problem is that the callback uri needs to be url encoded. 

A potential solution is found here:  https://developer.apple.com/forums/thread/679251?answerId=675057022#675057022

The crash looks like this:

```
2021-05-19 22:33:23.701893-0500 Runner[8651:75516] [AuthenticationSession] The provided scheme is not valid. A scheme should not include special characters such as ":" or "/".

2021-05-19 22:33:23.724882-0500 Runner[8651:75516] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'The provided scheme is not valid. A scheme should not include special characters such as ":" or "/".'

*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff20422fba __exceptionPreprocess + 242
	1   libobjc.A.dylib                     0x00007fff20193ff5 objc_exception_throw + 48
	2   AuthenticationServices              0x00007fff4aae1b15 -[ASWebAuthenticationSession initWithURL:callbackURLScheme:usingEphemeralSession:completionHandler:] + 684
	3   AWSMobileClient                     0x00000001012773eb -[AWSCognitoAuth launchASWebAuthenticationSession:] + 257
	4   AWSMobileClient                     0x000000010127703d -[AWSCognitoAuth launchSignInVC:] + 1589
	5   AWSMobileClient                     0x00000001012781fb -[AWSCognitoAuth getSessionInternal:completion:] + 1832
	6   AWSMobileClient                     0x0000000101276890 __47-[AWSCognitoAuth enqueueGetSession:completion:]_block_invoke + 43
```

**To Reproduce**
1. Call showSignIn method on AWSMobileClient, like this:

```
            let hostedUIOptions = HostedUIOptions(identityProvider: identityProvider, signInPrivateSession: true)
            AWSMobileClient.default().showSignIn(presentationAnchor: self.appDelegate.window,
                                                 hostedUIOptions: hostedUIOptions,
                                                 signInCompletionHandler)
```

**Observed Behavior**
Crash

**Expected Behavior**
Start web session for SSO sign in.

**Stack Trace**
```
	0   CoreFoundation                      0x00007fff20422fba __exceptionPreprocess + 242
	1   libobjc.A.dylib                     0x00007fff20193ff5 objc_exception_throw + 48
	2   AuthenticationServices              0x00007fff4aae1b15 -[ASWebAuthenticationSession initWithURL:callbackURLScheme:usingEphemeralSession:completionHandler:] + 684
	3   AWSMobileClient                     0x00000001012773eb -[AWSCognitoAuth launchASWebAuthenticationSession:] + 257
	4   AWSMobileClient                     0x000000010127703d -[AWSCognitoAuth launchSignInVC:] + 1589
	5   AWSMobileClient                     0x00000001012781fb -[AWSCognitoAuth getSessionInternal:completion:] + 1832
	6   AWSMobileClient                     0x0000000101276890 __47-[AWSCognitoAuth enqueueGetSession:completion:]_block_invoke + 43
	7   Foundation                          0x00007fff207fa0be __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 7
	8   Foundation                          0x00007fff207f9fb6 -[NSBlockOperation main] + 98
	9   Foundation                          0x00007fff207fcfd2 __NSOPERATION_IS_INVOKING_MAIN__ + 17
	10  Foundation                          0x00007fff207f9200 -[NSOperation start] + 785
	11  Foundation                          0x00007fff207fd947 __NSOPERATIONQUEUE_IS_STARTING_AN_OPERATION__ + 17
	12  Foundation                          0x00007fff207fd46a __NSOQSchedule_f + 182
	13  libdispatch.dylib                   0x00000001055a22c5 _dispatch_block_async_invoke2 + 83
	14  libdispatch.dylib                   0x000000010559374e _dispatch_client_callout + 8
	15  libdispatch.dylib                   0x0000000105596066 _dispatch_continuation_pop + 557
	16  libdispatch.dylib                   0x000000010559547b _dispatch_async_redirect_invoke + 770
	17  libdispatch.dylib                   0x00000001055a51b0 _dispatch_root_queue_drain + 351
	18  libdispatch.dylib                   0x00000001055a5b0f _dispatch_worker_thread2 + 135
	19  libsystem_pthread.dylib             0x00007fff6034047a _pthread_wqthread + 244
	20  libsystem_pthread.dylib             0x00007fff6033f493 start_wqthread + 15
```

**Code Snippet**
```
            let hostedUIOptions = HostedUIOptions(identityProvider: identityProvider, signInPrivateSession: true)
            AWSMobileClient.default().showSignIn(presentationAnchor: self.appDelegate.window,
                                                 hostedUIOptions: hostedUIOptions,
                                                 signInCompletionHandler)
```

**Areas of the SDK you are using (AWSMobileClient, Cognito, Pinpoint, IoT, etc)?**
AWSMobileClient, Cognito, Cognito Hosted UI with signInPrivateSession = true

**Environment(please complete the following information):**
 - SDK Version: 2.22.3
 - Dependency Manager: Cocoapods
 - Swift Version : 5
 - Xcode Version: 12.5

**Device Information (please complete the following information):**
 - Device: any
 - iOS Version: iOS 14.5
 - Specific to simulators: no

**Additional context**
Our callback URL scheme is celltraksd

Attaching screenshots of the AWS console showing the full callback URL.
<img width="538" alt="Screen Shot 2021-05-19 at 11 21 29 PM" src="https://user-images.githubusercontent.com/55109813/118918928-2dd17000-b8f9-11eb-9114-186a39a48d43.png">

**Comment by doc-rj-celltrak** [comment]
Looks like this was fixed in issue 1086 and release 2.23.1.  I'll leave this open until I can confirm.

**Comment by doc-rj-celltrak** [comment]
Closing, this was fixed in 2.23.1.
